src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCM.java

```diff
@@ -304,12 +305,13 @@ public FormValidation doCheckRepositorySlug(@QueryParameter String value) {
         @POST
         public FormValidation doCheckServerId(@QueryParameter String serverId) {
             Jenkins.get().checkPermission(Permission.CONFIGURE);
-            List<BitbucketServerConfiguration> serverList =
-                    new BitbucketPluginConfiguration().getServerList();
             // Users can only demur in providing a server name if none are available to select
-            if (!serverList.isEmpty() && isBlank(serverId)) {
+            if (bitbucketPluginConfiguration.getValidServerList().stream().noneMatch(server -> server.getId().equals(serverId))) {
                 return FormValidation.error("Please specify a valid Bitbucket Server.");
```

Comment:
I wonder if we should have a check earlier for invalid config so this message could inform the users that here is at least one invalid config, so maybe the server they're looking for is actually not configured properly? Thoughts?